core/modules/replica/HttpIngestConfigModule.cc

```diff
@@ -0,0 +1,179 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/HttpIngestConfigModule.h"
+
+// System headers
+#include <stdexcept>
+
+// Qserv headers
+#include "replica/Configuration.h"
+#include "replica/DatabaseServices.h"
+#include "replica/HttpExceptions.h"
+#include "replica/IngestConfigTypes.h"
+
+using namespace std;
+using json = nlohmann::json;
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+void HttpIngestConfigModule::process(Controller::Ptr const& controller,
+                                     string const& taskName,
+                                     HttpProcessorConfig const& processorConfig,
+                                     qhttp::Request::Ptr const& req,
+                                     qhttp::Response::Ptr const& resp,
+                                     string const& subModuleName,
+                                     HttpModule::AuthType const authType) {
+    HttpIngestConfigModule module(controller, taskName, processorConfig, req, resp);
+    module.execute(subModuleName, authType);
+}
+
+
+HttpIngestConfigModule::HttpIngestConfigModule(Controller::Ptr const& controller,
+                                               string const& taskName,
+                                               HttpProcessorConfig const& processorConfig,
+                                               qhttp::Request::Ptr const& req,
+                                               qhttp::Response::Ptr const& resp)
+    :   HttpModule(controller, taskName, processorConfig, req, resp) {
+}
+
+
+json HttpIngestConfigModule::executeImpl(string const& subModuleName) {
+    if (subModuleName == "GET") return _get();
+    else if (subModuleName == "UPDATE") return _update();
+    throw invalid_argument(
+            context() + "::" + string(__func__) +
+            "  unsupported sub-module: '" + subModuleName + "'");
+}
+
+
+json HttpIngestConfigModule::_get() {
+    debug(__func__);
+
+    auto const config = controller()->serviceProvider()->config();
+    auto const databaseServices = controller()->serviceProvider()->databaseServices();
+
+    auto const database = body().required<string>("database");
+    auto const databaseInfo = config->databaseInfo(database);
+
+    debug(__func__, "database=" + database);
+
+    // Ignore missing parameters if they were not set.
+
+    json result({{"database", databaseInfo.name}});
+
+    try {
+        result[HttpFileReaderConfig::sslVerifyPeerKey] = stoi(databaseServices->ingestParam(
+                databaseInfo.name,
+                HttpFileReaderConfig::category,
+                HttpFileReaderConfig::sslVerifyPeerKey).value);
+    } catch (DatabaseServicesNotFound const&) {}
```

Comment:
If missing parameter is not an exceptional case then raising an exception is potentially a problem. Maybe return a default value in that case would work better?

Reply:
This would violate to the expected contract of the REST service which is expected to return only those parameters that were explicitly set by a user, not any defaults. The very same logic (to do only what user explicitly asked for) is been following in the rest of the code, including a code configuring `libcurl` in the worker ingest service.
So, I politely refuse this suggestion 😸